packages/@sanity/cli/src/util/journeyConfig.ts

```diff
@@ -0,0 +1,264 @@
+        reject(new Error(`Worker stopped with exit code ${code}`))
+      }
+    })
+  })
+}
+
+/**
+ * Fetch a Journey config from the Sanity schema club API
+ *
+ * @param projectId - The slug of the Journey schema to fetch
+ * @returns The Journey schema as an array of Sanity document or object definitions
+ */
+export async function fetchJourneyConfig(
+  apiClient: CliApiClient,
+  projectId: string,
+): Promise<JourneyConfigResponse> {
+  if (!projectId) {
+    throw new Error('ProjectId is required')
+  }
+  if (!/^[a-zA-Z0-9-]+$/.test(projectId)) {
+    throw new Error('Invalid projectId')
+  }
+  try {
+    const response: {
+      projectId: string
+      dataset: string
+      displayName?: string
+      schemaUrl: string
+    } = await apiClient({
+      requireUser: true,
+      requireProject: true,
+      api: {projectId},
+    })
+      .config({apiVersion: 'v2024-02-23'})
+      .request({
+        method: 'GET',
+        uri: `/journey/projects/${projectId}`,
+      })
+
+    return {
+      projectId: response.projectId,
+      datasetName: response.dataset,
+      displayName: response.displayName || 'Sanity Project',
+      // The endpoint returns a signed URL that can be used to fetch the schema as ESM
+      schemaUrl: response.schemaUrl,
+      isFirstProject: true,
+    }
+  } catch (err) {
+    throw new Error(`Failed to fetch remote schema config: ${projectId}`)
+  }
+}
+
+/**
+ * Fetch a Journey schema from the Sanity schema club API
+ *
+ * @param projectId - The slug of the Journey schema to fetch
+ * @returns The Journey schema as an array of Sanity document or object definitions
+ */
+async function fetchJourneySchema(schemaUrl: string): Promise<DocumentOrObject[]> {
+  try {
+    const response = await import(schemaUrl)
+    return response.default
+  } catch (err) {
+    throw new Error(`Failed to fetch remote schema: ${schemaUrl}`)
+  }
+}
+
+/**
+ * Assemble a Journey schema type into a module export
+ * Include the necessary imports and export the schema type as a named export
+ *
+ * @param schema - The Journey schema to export
+ * @returns The Journey schema as a module export
+ */
+async function assembleJourneySchemaTypeFileContent(schemaType: DocumentOrObject): Promise<string> {
+  const serialised = wrapSchemaTypeInHelpers(schemaType)
+  const imports = getImports(serialised)
+  const prettifiedSchemaType = await format(serialised, {parser: 'typescript'})
+  // Start file with import, then export the schema type as a named export
+  return `${imports}\n\nexport const ${schemaType.name} = ${prettifiedSchemaType}\n`
+}
+
+/**
+ * Assemble a list of Journey schema module exports into a single index file
+ *
+ * @param schemas - The Journey schemas to assemble into an index file
+ * @returns The index file as a string
+ */
+function assembleJourneyIndexContent(schemas: DocumentOrObject[]): string {
+  schemas.sort((a, b) => (a.name > b.name ? 1 : -1)) // Sort schemas alphabetically by name
```

Comment:
For the sake of being defensive, could you avoid the in-place mutation using sort, and instead make a copy?
e.g `const sortedSchema = schemas.slice().sort((a, b) =>…)`